### DISCUSSION TITLE
OpenFAST v3.2.0

### DISCUSSION THREAD

**Discussion by rafmudaf** [OpenFAST/openfast#1179] [General]
# Release Notes

### General
#866 Include legacy documentation in pdf and MS Word format - Modules
#1021 Reg_tests python scripts: fix issue where directory returned is empty
#1062 Add documentation for community contribution
#1169 Update git-module urls
#1158 Update orientation differences in linear trim solution
#1177 Add documentation on 3D rotations in linearization
#1178 Corrections for 3.2.0 release

### AeroDyn
#932 BugFix: UA update states that were not updated
#1032 AD: added more nodal outputs for debugging UA
#1039 [BugFix] Nacelle motion not passed between AeroDyn driver and AeroDyn
#1043 AeroDyn: combine some FVW and BEMT output calculations
#1045 OLAF: check for division by zero to avoid invalid calculations
#1141 Move AD module reg tests to a standalone job

### BeamDyn
#1050 Fix aeroelastic stability analysis with BeamDyn

### FAST.Farm
#1107 Fix Time-Step Delay in Super Controller within FAST.Farm

### HydroDyn
#806 C-bindings interface for HydroDyn
#1047 Fix HydroDyn Jacobian outputs when LinOutJac is True
#1108 BugFix: Checks to populate matrix input to Newman's app were reversed
#1173 Bug Fix: rotation matrix perturbation with small angles was wrong

### InflowWind
#1144 BugFix: Fix typo in DLLEXPORT attributes for IfW C interface

### MAP++
#1048 MAP: small fix based on MAP 1.3
#1168 Fix MAP linearization operating point

### NWTC Library
#1050 Fix aeroelastic stability analysis with BeamDyn -- changes to the library routines for angle perturbations during linearization
#1124 Updates to VersionInfo module and its dependencies
#1157 Fix for binary file compression

### OpenFAST Library
#962 Add a Python glue-code interface
#1057 FAST Library: add access to hub position and velocity

### ServoDyn
#803 Linearization of ServoDyn Structural control elements
#1074 [BugFix] ServoDyn StC control signal channels were not zeroed properly
#1089 BugFix: Fix API change line numbers
#1101 BugFix: ServoDyn API change docs
#1140 Add non-rotating hub forces to Bladed Interface
#1160 Correction to documentation on Extended Bladed DLL interface

### TurbSim
#887 TurbSim modifications

# Precompiled binaries

Coming soon

<hr /><em>This discussion was created from the release <a href='https://github.com/OpenFAST/openfast/releases/tag/v3.2.0'>OpenFAST v3.2.0</a>.</em>

**Comment by sautee** [comment]
Hello,

I use Windows, and I'm trying to run openfast_x64.exe with a NREL-5MW.fst file. Attached [NREL-5MW_fst.txt](https://github.com/OpenFAST/openfast/files/9188707/NREL-5MW_fst.txt)

It starts to run the simulation, but it fails looking for the **libdiscon.so** file. Since I'm running in Windows, shouldn't it be looking for **libdiscon.dll** file instead? Here's the full error

```
OpenFAST-v3.2.0
Compile Info:
 - Compiler: Intel(R) Fortran Compiler 1900
 - Architecture: 64 bit
 - Precision: single
 - OpenMP: No
 - Date: Jul 13 2022
 - Time: 09:58:55
Execution Info:
 - Date: 07/26/2022
 - Time: 07:16:49-0230

OpenFAST input file heading:
    FAST Certification Test #18: NREL 5.0 MW Baseline Wind Turbine (Onshore)

Running ElastoDyn.
Nodal outputs section of ElastoDyn input file not found or improperly formatted.
Running AeroDyn.
AD15 Nodal Outputs: Nodal output section of AeroDyn input file not found or improperly formatted.
Skipping nodal outputs.
Warning: Turning off Unsteady Aerodynamics because UA parameters are not included in airfoil
(airfoil has likely has constant polars). (node 1, blade 1)
Warning: Turning off Unsteady Aerodynamics because UA parameters are not included in airfoil
(airfoil has likely has constant polars). (node 2, blade 1)
Warning: Turning off Unsteady Aerodynamics because UA parameters are not included in airfoil
(airfoil has likely has constant polars). (node 3, blade 1)
Warning: Turning off Unsteady Aerodynamics because UA parameters are not included in airfoil
(airfoil has likely has constant polars). (node 4, blade 1)
Warning: Turning off Unsteady Aerodynamics because UA parameters are not included in airfoil
(airfoil has likely has constant polars). (node 1, blade 2)
Warning: Turning off Unsteady Aerodynamics because UA parameters are not included in airfoil
(airfoil has likely has constant polars). (node 2, blade 2)
Warning: Turning off Unsteady Aerodynamics because UA parameters are not included in airfoil
(airfoil has likely has constant polars). (node 3, blade 2)
Warning: Turning off Unsteady Aerodynamics because UA parameters are not included in airfoil
(airfoil has likely has constant polars). (node 4, blade 2)
Warning: Turning off Unsteady Aerodynamics because UA parameters are not included in airfoil
(airfoil has likely has constant polars). (node 1, blade 3)
Warning: Turning off Unsteady Aerodynamics because UA parameters are not included in airfoil
(airfoil has likely has constant polars). (node 2, blade 3)
Warning: Turning off Unsteady Aerodynamics because UA parameters are not included in airfoil
(airfoil has likely has constant polars). (node 3, blade 3)
Warning: Turning off Unsteady Aerodynamics because UA parameters are not included in airfoil
(airfoil has likely has constant polars). (node 4, blade 3)
Running InflowWind.
Running ServoDyn.
Running ServoDyn Interface for Bladed Controllers (using Intel Visual Fortran for Windows).

FAST_InitializeAll:SrvD_Init:BladedInterface_Init:The dynamic library
C:\Users\andre\rosco-test\Test_Cases\NREL-5MW\../../ROSCO/install/lib/libdiscon.so could not be
loaded. Check that the file exists in the specified location and that it is compiled for 64-bit
applications.

 OpenFAST encountered an error during module initialization.
 Simulation error level: FATAL ERROR

 Aborting OpenFAST.
```

**Reply by rafmudaf** [reply]
You can change the path to the library in the ServoDyn input file. In your case, it is `NRELOffshrBsline5MW_Onshore_ServoDyn.dat` alongside your main OpenFAST input (`.fst`).

**Reply by sautee** [reply]
Perfect, that worked. Thank you for the quick assist!
